lib/vsc/administration/slurm/sync.py

```diff
@@ -274,20 +306,38 @@ def slurm_vo_accounts(account_page_vos, slurm_account_info, clusters, host_insti
     """
     commands = []
     for cluster in clusters:
-        cluster_accounts = [acct.Account for acct in slurm_account_info if acct and acct.Cluster == cluster]
+        cluster_accounts = dict([
+            (acct.Account, int(acct.Share))
+            for acct in slurm_account_info
+            if acct and acct.Cluster == cluster
+        ])
 
         for vo in account_page_vos:
+
+            # skip the "default" VOs for our own institute
             if vo.vsc_id in INSTITUTE_VOS_BY_INSTITUTE[host_institute].values():
                 continue
 
+            # create a new account for a VO that does not already have an account
             if vo.vsc_id not in cluster_accounts:
                 commands.append(create_add_account_command(
                     account=vo.vsc_id,
                     parent=vo.institute['name'],
                     cluster=cluster,
-                    organisation=vo.institute['name']
+                    organisation=vo.institute['name'],
+                    fairshare=vo.fairshare,
+                ))
+
+            # create update commands for VOs with a changed fairshare
+            elif int(vo.fairshare) != cluster_accounts[vo.vsc_id]:
```

Comment:
why is the `int()` needed? how sure are we that `cluster_accounts[vo.vsc_id]` are int?

Reply:
Well. If we'd have a decent static type system, we'd know for sure :D 

~~~~
{
  "vsc_id": "gvo00002",
  "status": "active",
  "vsc_id_number": 2640009,
  "institute": {
    "name": "gent"
  },
  "fairshare": 100,
~~~~

But it does seem to be an actual int in the returned JSON (because of course it is). 

As for the `cluster_accounts[vo.vsc_id] ... 

![image](https://user-images.githubusercontent.com/196619/119890478-2e40bb00-bf38-11eb-850f-726ec96a73b1.png)

So PyLance's type checker is quite sure it's an `int`.